## Short read assembly

#### Read correction using `bfc`
Read correction is useful to remove sequencing errors from sequencing reads, thus facilitating efficiency and accuraycy of the genome assembly. 
```ShellSession
bfc -b 32 -k 25 -t 8 S1529Nr1_R1_trim.fq.gz | gzip -1 > S1529Nr1_R1_trim.corrected.fq.gz
bfc -b 32 -k 25 -t 8 S1529Nr1_R2_trim.fq.gz | gzip -1 > S1529Nr1_R2_trim.corrected.fq.gz
bfc -b 32 -k 25 -t 8 S1529Nr1_unpaired_trim.fq.gz | gzip -1 > S1529Nr1_unpaired_trim.corrected.fq.gz
```

#### Assembly with `SPAdes`
If only Illumina short-read sequencing data is available, a short-read assembler is required. One option I used is `SPAdes`:

```ShellSession
spades.py -1 read_1_trimmed.corrected.fq.gz -2 read_2_trimmed.corrected.fq.gz -s reads_unpaired_trimmed.corrected.fq.gz --only-assembler -k 31,51,71,91,121 -o new_assembly
```

#### Assembly with `SOAPdenovo2`
Another much faster and more efficient option is `SOAPdenovo2`. `SOAPdenovo2` requires a config-file to run, which looked like this: 
```
#maximal read length
max_rd_len=150
[LIB]
#average insert size
avg_ins=300
#if sequence needs to be reversed 
reverse_seq=0
#in which part(s) the reads are used
asm_flags=3
#use only first 100 bps of each read
rd_len_cutoff=150
#in which order the reads are used while scaffolding
rank=1
# cutoff of pair number for a reliable connection (at least 3 for short insert size)
pair_num_cutoff=3
#minimum aligned length to contigs for a reliable read location (at least 32 for short insert size)
map_len=32
#a pair of fastq file, read 1 file should always be followed by read 2 file
q1=/hpcwork/sk893857/DNA-Seq/trimmed/Aus1_R1_trim.corrected.fq.gz
q2=/hpcwork/sk893857/DNA-Seq/trimmed/Aus1_R2_trim.corrected.fq.gz
#fastq file for single reads
q=/hpcwork/sk893857/DNA-Seq/trimmed/Aus1_R1_trim_unpaired.corrected.fq.gz
```
Then, I ran the assembler: 
```ShellSession
SOAPdenovo-127mer all -p 48 -s config-file -K 127 -R -N 120000000 -o assembly-directory 1>assembly.log 2>assembly.err
```
`SOAPdenovo2` offers two binaries (-127mer and -63mer) optimized for different maximum kmers. `-R` means resolve repeats by reads. `-N 120000000` indicates the approximate genome size. `-K 127` sets the kmer to 127. 
