#### Re-mapping short reads
It can be useful to re-map long reads from PacBio or nanopore to assess the coverage of contigs. `hisat2` is a tool that allows this:
```ShellSession
hisat2-build Ltaurica_exCapsicumHu_v0.fa Ltaurica-v0_index #for the index
hisat2 -p 8 --no-spliced-alignment -k 10 -x Ltaurica-v0_index -1 S1529Nr1_R1_trim.fq.gz -2 S1529Nr1_R2_trim.fq.gz -U S1529Nr1_R1_trim_uF.fq.gz -S S1529Nr1_hisat.sam
samtools sort -l 0 -o S1529Nr1_hisat.bam -O bam S1529Nr1_hisat.sam #create the BAM file
samtools view -F 0x04 -b S1529Nr1_hisat.bam > S1529Nr1_hisat.aligned.bam #clean up BAM file from non-mapping reads
```

#### BLASTN against NCBI nt database
Using `NCBI BLAST+ 2.7` and the latest version of the NCBI nucleotide collection database nt, I aligned the contigs against the known nucleotide sequences. This can help identify contigs of other origin (contaminant, viruses, mitochondrial genome). 
```ShellSession
blastn -task megablast -query Ltaurica_exCapsicumHu_v0.fa -db $HOME/blastdb/nt -outfmt '6 qseqid staxids bitscore std sscinames sskingdoms stitle' -culling_limit 5 -num_threads 32 -evalue 1e-25 -out Ltaurica_exCapsicumHu_v0.blast-nt.tbl
```

#### Visualizing data with `Blobtools`
`Blobtools` is a very useful tool to summarize and visualize BLAST-nt and coverage data, and helps rapidly identify contaminant contigs. The putative identity of contigs can be generated on the levels of superkingdom, class, phylum, family, and genus.
```ShellSession
blobtools create -i Ltaurica_exCapsicumHu_v0.fa -b S1529Nr1_hisat.aligned.bam -t Ltaurica_exCapsicumHu_v0.blast-nt.tbl -o blobfile --db $HOME/miniconda3/opt/blobtools-1.0.1/data/nodesDB.txt #creates the JSON file
blobtools blobplot -i blobfile.blobDB.json -o Ltaurica-v0_plot -r family --format pdf #to visualize as PDF
blobtools view -i blobfile.blobDB.json -r family -o Ltaurica-v0_table #the table helps to inspect statistics (coverage, identity, GC content) of contigs
```

This is the table:
```
## blobtools v1.0
## assembly	: Ltaurica_exCapsicumHu_v0.fa
## coverage	: bam0 - S1529Nr1_hisat.aligned.bam
## taxonomy	: tax0 - Ltaurica_exCapsicumHu_v0.blast-nt.tbl
## nodesDB	: $HOME/miniconda3/opt/blobtools-1.0.1/data/nodesDB.txt
## taxrule	: bestsum
## min_score	: 0.0
## min_diff	: 0.0
## tax_collision_random	: False
##
# name	length	GC	N	bam0	family.t.6	family.s.7	family.c.8
scaffold_00001	1008	0.4547	58	584.053	no-hit	0.0	0
scaffold_00003	15245	0.386	363	15.126	no-hit	0.0	0
scaffold_00006	2621	0.3757	207	16.079	no-hit	0.0	0
scaffold_00009	1848	0.3727	182	130.892	Albuginaceae	588.0	0
scaffold_00011	2005	0.4194	174	12.894	no-hit	0.0	0
scaffold_00012	2725	0.3957	155	10.276	no-hit	0.0	0
scaffold_00013	1180	0.3701	56	14.294	Erysiphaceae	290.0	0
scaffold_00014	11099	0.3706	78	14.31	no-hit	0.0	0
scaffold_00017	24492	0.3945	203	12.118	Erysiphaceae	755.0	4
scaffold_00021	5282	0.4026	207	27.64	Albuginaceae	3761.0	0
scaffold_00022	1004	0.3213	36	34.613	no-hit	0.0	0
scaffold_00023	38175	0.3904	231	13.63	Erysiphaceae	2572.0	1
scaffold_00024	13584	0.386	388	17.328	Erysiphaceae	13473.0	0
scaffold_00025	47872	0.3953	396	12.989	Erysiphaceae	176.0	0
```

#### Filtering by GC, coverage and blast identity
Using the table from `BlobTools`, I pre-filtered the list for scaffolds/contigs with the identity *Erysiphaceae*, *Albuginaceae*, or no-hit. 
```ShellSession
echo -e "Erysiphaceae \nAlbuginaceae" | grep -Fwf - Ltaurica-v0_table.family.blobDB.table.txt > list.contigs.out
grep no-hit Ltaurica-v0_table.family.blobDB.table.txt | awk '(($3 >= 0.35) && ($3 <= 0.55))' > list.GC35-55.out #only keep no-hit contigs with a GC content of 0.35-0.55
grep no-hit list.GC35-55.out | awk '($5 > 50)' > list.GC35-55_cov50.out #only retain no-hit contigs with mapping coverage >50
cat list.contigs.out list.GC35-55_cov50.out > list.contigs.GC35-55_cov50.out #the list of candidate powdery mildew contigs
###to create the fasta file
seqkit grep -n -f list.contigs.GC35-55_cov50.out Ltaurica_exCapsicumHu_v0.fa > Ltaurica_exCapsicumHu_v1.fa
```
The draft assembly `Ltaurica_exCapsicumHu_v1.fa` was then tested with this procedure again for confirmation. 
