#### Genome completeness 
A very useful measure of genome quality is to use `BUSCO v3`. This algorithm searches for core conserved gene sets of lineages to assess the completeness of a given genome or proteome.  

```ShellSession
python $HOME/miniconda3/bin/BUSCO.py -f --out Ltaurica_v1.busco --in Ltaurica_exCapsicumHu_v1.fasta --lineage ~/database/BUSCO/ascomycota_odb9 -m genome
```

The summary output of `BUSCO` reveals the number of complete, fragmented, duplicated, and missing core genes of the selected lineage, here ascomycota_odb9.
```
C:85.0%[S:85.0%,D:0.0%],F:1.3%,M:13.7%,n:1315

	1118	Complete BUSCOs (C)
	1118	Complete and single-copy BUSCOs (S)
	0	Complete and duplicated BUSCOs (D)
	17	Fragmented BUSCOs (F)
	180	Missing BUSCOs (M)
	1315	Total BUSCO groups searched
```

#### Repeat content
`RepeatMasker` allows the identification of known transposable elements and simple repetitive DNA. It also masks repetitive sequences.
```ShellSession
RepeatMasker -pa 8 -species fungi -gff -dir $HOME/RepeatMasker Mysul_MySu01_v1.fa
```
Note that I ran `RepeatMasker` with RepBase-20170127 and using version 4.0.7, since I lack access to later RepBase versions. 

`RepeatMasker` produces a GFF file of repeat elements, a genome file where repetitive elements are masked, and an output table looking like this:
```
==================================================
file name: Ltaurica_exCapsicumHu_v1.fasta
sequences:         24103
total length:  188302910 bp  (186289740 bp excl N/X-runs)
GC level:         39.12 %
bases masked:   57051256 bp ( 30.30 %)
==================================================
               number of      length   percentage
               elements*    occupied  of sequence
--------------------------------------------------
Retroelements        63746     47431049 bp   25.19 %
   SINEs:             3249       632967 bp    0.34 %
   Penelope              2           86 bp    0.00 %
   LINEs:            10051      8186492 bp    4.35 %
    CRE/SLACS            0            0 bp    0.00 %
     L2/CR1/Rex          0            0 bp    0.00 %
     R1/LOA/Jockey      63         8332 bp    0.00 %
     R2/R4/NeSL          0            0 bp    0.00 %
     RTE/Bov-B           2           96 bp    0.00 %
     L1/CIN4             0            0 bp    0.00 %
   LTR elements:     50446     38611590 bp   20.51 %
     BEL/Pao             0            0 bp    0.00 %
     Ty1/Copia       20260     15973826 bp    8.48 %
     Gypsy/DIRS1     29641     22549680 bp   11.98 %
       Retroviral        0            0 bp    0.00 %

DNA transposons      14946      3624293 bp    1.92 %
   hobo-Activator        6          542 bp    0.00 %
   Tc1-IS630-Pogo    13746      3482056 bp    1.85 %
   En-Spm                0            0 bp    0.00 %
   MuDR-IS905            0            0 bp    0.00 %
   PiggyBac              8         1684 bp    0.00 %
   Tourist/Harbinger    83        10241 bp    0.01 %
   Other (Mirage,        2           71 bp    0.00 %
    P-element, Transib)

Rolling-circles          0            0 bp    0.00 %

Unclassified:        14525      4508258 bp    2.39 %

Total interspersed repeats:    55563600 bp   29.51 %


Small RNA:              19         7494 bp    0.00 %

Satellites:             15         1125 bp    0.00 %
Simple repeats:      24840      1322782 bp    0.70 %
Low complexity:       3108       157190 bp    0.08 %
==================================================

* most repeats fragmented by insertions or deletions
  have been counted as one element
                                                      

The query species was assumed to be fungi         
RepeatMasker Combined Database: Dfam_Consensus-20170127, RepBase-20170127
        
run with rmblastn version 2.6.0+
```
